src/foam/box/SessionServerBox.java

```diff
@@ -123,6 +128,24 @@ public void send(Message msg) {
         return;
       }
 
+      // Decorate the reply box to add the sessionId as an attribute to the
```

Comment:
@KernelDeimos since we talked about you doing something similar, here's an example of adding a replyBox decorator on the server side.